src/GovUk.Education.ExploreEducationStatistics.Data.Model/Services/ReleaseSubjectService.cs

```diff
@@ -18,77 +18,83 @@ public ReleaseSubjectService(StatisticsDbContext statisticsDbContext, IFootnoteS
             _footnoteService = footnoteService;
         }
 
-        public async Task SoftDeleteAllSubjectsOrBreakReleaseLinks(Guid releaseId)
+        public async Task SoftDeleteAllReleaseSubjects(Guid releaseId)
         {
-            await DeleteAllSubjectsOrBreakReleaseLinks(releaseId, true);
+            await DeleteAllReleaseSubjects(releaseId, true);
         }
 
-        public async Task SoftDeleteSubjectOrBreakReleaseLink(Guid releaseId, Guid subjectId)
+        public async Task SoftDeleteReleaseSubject(Guid releaseId, Guid subjectId)
         {
-            await DeleteSubjectOrBreakReleaseLink(releaseId, subjectId, true);
+            await DeleteReleaseSubject(releaseId, subjectId, true);
         }
 
-        public async Task DeleteAllSubjectsOrBreakReleaseLinks(Guid releaseId, bool isSoftDelete = false)
+        public async Task DeleteAllReleaseSubjects(Guid releaseId, bool softDeleteOrphanedSubjects = false)
         {
             var subjectIds = await _statisticsDbContext.ReleaseSubject
                 .Include(rs => rs.Subject)
-                .Where(rs => rs.ReleaseId == releaseId).Select(rs => rs.SubjectId).ToListAsync();
+                .Where(rs => rs.ReleaseId == releaseId)
+                .Select(rs => rs.SubjectId)
+                .ToListAsync();
 
             foreach (var id in subjectIds)
             {
-                await DeleteSubjectOrBreakReleaseLink(releaseId, id, isSoftDelete);
+                await DeleteReleaseSubject(releaseId, id, softDeleteOrphanedSubjects);
             }
         }
 
-        public async Task DeleteSubjectOrBreakReleaseLink(Guid releaseId, Guid subjectId, bool isSoftDelete = false)
-        {
-            await RemoveReleaseSubjectLinkIfExists(releaseId, subjectId);
-            await _footnoteService.DeleteAllFootnotesBySubject(releaseId, subjectId);
-            await DeleteSubjectIfOrphanedAndExists(subjectId, isSoftDelete);
-
-            await _statisticsDbContext.SaveChangesAsync();
-        }
-
-        private async Task RemoveReleaseSubjectLinkIfExists(Guid releaseId, Guid subjectId)
+        public async Task DeleteReleaseSubject(Guid releaseId, Guid subjectId, bool softDeleteOrphanedSubject = false)
         {
             var releaseSubject = await _statisticsDbContext
                 .ReleaseSubject
+                .Include(rs => rs.Subject)
                 .FirstOrDefaultAsync(rs => rs.ReleaseId == releaseId && rs.SubjectId == subjectId);
 
+            await DeleteReleaseSubjectIfExists(releaseSubject);
+            await _footnoteService.DeleteAllFootnotesBySubject(releaseId, subjectId);
+            await DeleteSubjectIfOrphaned(releaseSubject.Subject, softDeleteOrphanedSubject);
```

Comment:
The saving logic is a bit strange now. Previously there was one save at the end so the `ReleaseSubject` would still exist if there was an exception thrown deleting the Footnotes or the Subject. Now, the first and third calls do their own save but `DeleteAllFootnotesBySubject` doesn't. Just wondering if you had spotted this?

Reply:
Nope I hadn't, mostly as I would have expected `FootnoteService` to do its own saving! I think I'll just add an additional save to `FootnoteService`.

Reply:
Cool.